tensorflow/compiler/jit/extract_outside_compilation_pass.cc

```diff
@@ -1240,11 +1241,11 @@ Status RewriteShapeInferenceGraph(const string& shape_inference_graph_name,
     // This is an "top-level" outside compilation. Clear the graph, and copy
     // SendFromHost and all its predecessors from `host_graph`.
     std::vector<Node*> nodes;
```

Comment:
@er91 So you want me to remove these lines, replacing the last with `for (Node* n : g->op_nodes())`:
```c++
const auto op_nodes = g->op_nodes();
nodes.reserve(std::distance(op_nodes.begin(), op_nodes.end()));
for (Node* n : op_nodes) {
```

Reply:
No you shouldn't do `for (Node* n : g->op_nodes())` either, because in the loop we remove node from `g>op_nodes()`. In general we should not mutate a vector/list/container while iterating on it.

One thing you can do, to match your intention of "call reserve() for vectors when we know the vector size upfront", is adding `nodes.reserve(g->num_op_nodes())`.

Reply:
Ahh yep, that's the one. Thanks. Fixed.

(I just grokked the codebase looking for looping over vectors with push back every iteration)